src/main/java/io/getunleash/util/UnleashConfig.java

```diff
@@ -567,6 +589,16 @@ private String getBackupFile() {
             }
         }
 
+        /**
+         * Adds a custom http header for authorizing the client
+         * @param apiKey the client key to use to connect to the Unleash Server
+         * @return
+         */
+        public Builder apiKey(String apiKey) {
+            this.customHttpHeaders.put("Authorization", apiKey);
```

Comment:
Nice, this is so much cleaner! Looks like this should also work on top of custom header providers without problems. 

Right now the examples on the README around headers still show the old usage where you'd need to inject the auth header manually, should we also update those to reflect this?

Reply:
Feels like a good idea. I'll update this.